app/controllers/library_hours_controller.rb

```diff
@@ -1,24 +1,108 @@
 # frozen_string_literal: true
 
 class LibraryHoursController < ApplicationController
+  before_action :get_locations, only: [:index, :show]
+  before_action :set_location, only: [:show]
+  before_action :set_dates, only: [:index, :show]
+
   def index
-    @locations = LibraryHours.all.pluck(:location_id).uniq
+    @buildings = [
+      {
+        slug: "paley",
+        spaces: [
+                  "paley",
+                  "media",
+                  "doc_del",
+                  "ref_desk",
+                  "v_ref",
+                  "thinktank",
+                  "scrc",
+                  "dsc",
+                  "guest_computers"
+                ]
+      },
+      {
+        slug: "ginsburg",
+        spaces: ["ginsburg", "innovation"]
+      },
+      {
+        slug: "podiatry",
+        spaces: ["podiatry"]
+      },
+      {
+        slug: "ambler",
+        spaces: ["ambler"]
+      },
+      {
+        slug: "blockson",
+        spaces: ["blockson"]
+      }
+    ]
+    @buildings.each do |building|
+      building.values.second.map! do |space|
+        space = [building.values.first, LibraryHours.where(location_id: space, date: @sunday..@saturday)]
+      end
+    end
   end
-  def show
-    @locations = LibraryHours.distinct.pluck(:location_id)
+
+  def set_dates
     @today = Date.today
-    sunday = @today.beginning_of_week - 1
-    saturday = @today.end_of_week
+    @cyear = @today.year
+    @cweek = @today.cweek
+
+    @week =  params[:week].nil? ? @cweek : params[:week].to_i
+    @year = params[:year].nil? ? @cyear : params[:year].to_i
+
+    @first_week = LibraryHours.where(location_id: "paley").first.date.to_date.cweek
+    @first_year = LibraryHours.where(location_id: "paley").first.date.to_date.year
+    @last_week = LibraryHours.where(location_id: "paley").last.date.to_date.cweek
+    @last_year = LibraryHours.where(location_id: "paley").last.date.to_date.year
+
+    @next_week = @week + 1
+    @prev_week = @week - 1
+
+    if @week == 52
```

Comment:
it feels like a lot of this could be handled by rails built in date extensions, like `.next_week`

```
t = Date.parse("31/12/2018")
=> Mon, 31 Dec 2018
t.next_week
=> Mon, 07 Jan 2019
```